lib/services/google_analytics.rb

```diff
@@ -0,0 +1,63 @@
+require 'google/apis/analyticsreporting_v4'
+require 'googleauth'
+
+module Services
+  class GoogleAnalytics
+    include Google::Apis::AnalyticsreportingV4
+    include Google::Auth
+
+    def client(scope: "https://www.googleapis.com/auth/analytics.readonly")
+      @client ||= AnalyticsReportingService.new
+      @client.authorization ||= ServiceAccountCredentials.make_creds(
+        json_key_io: File.open(ENV["GOOGLE_AUTH_CREDENTIALS"]),
+        scope: scope
+      )
+      @client
+    end
+
+    def page_views(base_paths)
+      raise "base_paths isn't an array" unless base_paths.is_a?(Array)
```

Comment:
What about extracting classes for parsing the request and the response?. Something like:

```ruby
request = GoogleAnalytics::PageViewsRequest.new
response = client.batch_get_reports(request.build)
GoogleAnalytics::PageViewsResponse.new(response).value
```

I see a few benefits:

1. The service is easier to test because there is no need to test requests and responses 
2. The requests and the responses can be tested in isolation.
3. It is a bit more aligned with the SRP.

We would need an integration test, to be sure that all the pieces are assembled properly.